src/components/header.jsx

```diff
@@ -49,6 +52,52 @@ function Header({ slug }) {
     ),
   );
 
+  const handleClick = (langKey) => {
+    i18n.changeLanguage(langKey);
+    navigate(`/${langKey}/${slug}`);
+  };
+
+  const languageNavItems = (
+    <div className="header__language-nav-container">
+      <div className="header__language-nav-RtoL">
+        {languages
+          .filter((language) => language.isRtoL)
```

Comment:
Do you have to filter then map? I think you can use a conditional in the map instead. It might look more readable the way you have it now, but you're doing more operations currently.

Reply:
Changed to conditional in map